Some changes and enhancements

teddyxlandlee opened the pull request:
SplashScreen -> MojangLogoLoadingScreenOverlay
class_5493 -> util/MobUtils

Commit:
SplashScreen -> MojangLogoLoadingScreenOverlay
class_5493 -> util/MobUtils

ramidzkh commented on mappings/net/minecraft/client/gui/screen/MojangLogoLoadingScreenOverlay.mapping:
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_425 net/minecraft/client/gui/screen/SplashScreen
+CLASS net/minecraft/class_425 net/minecraft/client/gui/screen/MojangLogoLoadingScreenOverlay
```
Not a fan of such a verbose name. `LoadingScreenOverlay` is enough

teddyxlandlee replied:
@ramidzkh What about the others? Do I need to reopen a PR?

liach replied:
You can just go back to your local yarn, rename this class, save, and use git to commit and push to your branch

pluiedev replied:
+1 for `LoadingScreenOverlay` (that was called a `Screen` even though it doesn't inherit from it; it literally wasted half an hour of my life trying to find it :sadtater:)

liach replied:
Should probably be like `SplashOverlay` then, see https://github.com/FabricMC/yarn/pull/1992#issuecomment-770912109

liach commented on mappings/net/minecraft/util/MobUtils.mapping:
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_5493 net/minecraft/util/MobUtils
```
Hmm, we usually only put really generic, i.e. not-that-much game related stuff in util. This thing we usually put in the entity package. In addition, I guess we might come up with a more specific name?

teddyxlandlee replied:
Hmm... I guess I have to put it in `net.minecraft.entity.mob` package and rename it. Let me take a century 😄   
Plus. What about the `LoadingScreenOverlay`? I have just pushed it on.

liach replied:
Hmm, I've named this class in #2030. I think my pr is more comprehensive as it covers the related code as a system.

Commit:
Update mappings
SplashScreen -> LoadingScreenOverlay
class_5493 -> MobUtils
  method_30955 -> withValidNavigation

liach commented on mappings/net/minecraft/entity/EntityManager.mapping:
```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/class_5569
+CLASS net/minecraft/class_5569 net/minecraft/entity/EntityManager
```
Lol no, this is an entity listener/callback assigned by the manager to each entity.
See https://github.com/FabricMC/yarn/pull/1967/files#diff-ef57025da02812a1e3d26e02f5612f228e64800e7afc5a84ef00d55818833599R1

liach commented:
Wait, does anyone know why the popup loading screen was named "splash"? I think that's because it pops up, right? Then SplashScreen is a better name. Fyi it's from #180

liach commented:
Also, look at https://www.dictionary.com/browse/splash?s=t which defines splash as a patch of color or light and a striking impression, I think the name splash should be kept.

Juuxel commented:
@liach It's also a *splash screen* with its own definition (see [Wikipedia](https://en.wikipedia.org/wiki/Splash_screen) or [JDK docs](https://docs.oracle.com/en/java/javase/15/docs/api/java.desktop/java/awt/SplashScreen.html))

liach commented:
For your 4 changes:
classes 5565 and 5569 and the fields you've mapped are already mapped in 47bdb3b9e45e939fb9adf618d062fe6268f66eca
class_5493 is part of #2030 which maps the whole path targeting logic more comprehensively
now this pr is really only for the rename of SplashScreen.

YanisBft commented:
Closing as this is outdated
